forage-android/src/main/java/com/joinforage/forage/android/ecom/ui/vault/forage/RosettaPinElement.kt

```diff
@@ -0,0 +1,208 @@
+package com.joinforage.forage.android.ecom.ui.vault.forage
+
+import android.content.Context
+import android.content.res.TypedArray
+import android.graphics.Color
+import android.graphics.Typeface
+import android.graphics.drawable.GradientDrawable
+import android.text.InputFilter
+import android.text.InputType
+import android.util.AttributeSet
+import android.util.TypedValue
+import android.view.Gravity
+import android.view.ViewGroup
+import android.view.inputmethod.InputMethodManager
+import android.widget.EditText
+import android.widget.LinearLayout
+import androidx.core.content.getSystemService
+import com.joinforage.forage.android.R
+import com.joinforage.forage.android.core.services.VaultType
+import com.joinforage.forage.android.core.services.telemetry.Log
+import com.joinforage.forage.android.core.services.vault.AbstractVaultSubmitter
+import com.joinforage.forage.android.core.ui.VaultWrapper
+import com.joinforage.forage.android.core.ui.element.ForagePinElement
+import com.joinforage.forage.android.core.ui.element.state.PinElementStateManager
+import com.joinforage.forage.android.core.ui.getBoxCornerRadiusBottomEnd
+import com.joinforage.forage.android.core.ui.getBoxCornerRadiusBottomStart
+import com.joinforage.forage.android.core.ui.getBoxCornerRadiusTopEnd
+import com.joinforage.forage.android.core.ui.getBoxCornerRadiusTopStart
+import com.joinforage.forage.android.core.ui.textwatcher.PinTextWatcher
+import com.joinforage.forage.android.ecom.services.vault.forage.RosettaPinSubmitter
+
+internal class RosettaPinElement @JvmOverloads constructor(
```

Comment:
@devinmorgan can this file be moved to `core` instead of `ecom` ?

This can also happen in a future PR as this is an internal change, but vocalizing now for visibility.

Reply:
The main blocker right now is that this file depends on:

`import com.joinforage.forage.android.ecom.services.vault.forage.RosettaPinSubmitter`

which is an `ecom` file. So in theory, yes, this file ought to live in `core` since it's used in both Ecom and Pos but this may be more appropriate to do in a follow on PR since it's slightly move inovled. Nice callout!